deployments/extensions.go

```diff
@@ -0,0 +1,405 @@
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	descFilePath := ""
+
+	if u.IsAbs() && u.Scheme != "" && u.Host != "" {
+		// "/" at the end is necessary so that last part of the path is not replaced
+		u, _ = u.Parse(e.Name + "/")
+		u, _ = u.Parse(defaultDescriptionFileName)
+		if err := helpers.DownloadFile(u.String(), defaultDescriptionFileName, tmpDir); err != nil {
+			return err
+		}
+		descFilePath = filepath.Join(tmpDir, defaultDescriptionFileName)
+	} else {
+		info, err := os.Stat(e.Repository)
+		if os.IsNotExist(err) {
+			return err
+		}
+		if !info.IsDir() {
+			return errors.New("Provided path to extension repository is neither URL nor a directory")
+
+		}
+		descFilePath = filepath.Join(e.Repository, e.Name, defaultDescriptionFileName)
+	}
+
+	// parse and load descriptin file into Extension struct
+	data, err := os.ReadFile(descFilePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to read description file")
+	}
+
+	err = yaml.Unmarshal(data, &e.desc)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse description file")
+	}
+	return nil
+}
+
+// Pass the path string and return the absolute location of the file
+// If the path is relative, join it with the base repository path; if
+// the path is URL download it and return path to downloaded copy
+func (e *Extension) fetchFile(filePath, tmpDir string) (string, error) {
+
+	// 1, local path is absolute, return right away
+	if filepath.IsAbs(filePath) {
+		return filePath, nil
+	}
+
+	name := filepath.Base(filePath)
+	u, err := url.Parse(filePath)
+	if err != nil {
+		return "", err
+	}
+	// 2. full URL, download and return path to copy
+	if u.IsAbs() && u.Host != "" {
+		if err := helpers.DownloadFile(u.String(), name, tmpDir); err != nil {
+			return "", err
+		}
+		return filepath.Join(tmpDir, name), nil
+	}
+	// 3. relative path to extension URL: adapt URL and download
+	u, err = url.Parse(e.Repository)
+	if u.IsAbs() && u.Host != "" {
+		u, _ = u.Parse(e.Name + "/")
+		u, _ = u.Parse(filePath)
+		if err := helpers.DownloadFile(u.String(), name, tmpDir); err != nil {
+			return "", err
+		}
+		return filepath.Join(tmpDir, name), nil
+	}
+	// 4. relative path to extension local path
+	return filepath.Join(e.Repository, e.Name, filePath), nil
+}
+
+func (e *Extension) executeScript(path string) error {
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	fullCmd, err := e.fetchFile(path, tmpDir)
+	if err != nil {
+		return errors.Wrap(err, "failed fetching file from "+path)
+	}
+
+	if out, err := helpers.RunProc(fullCmd, tmpDir, e.Debug); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed running script: %s\n", out))
+	}
+
+	return nil
+}
+
+func (e *Extension) installManifest(path, ns string) error {
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
```

Comment:
here you are using the same temp directory name as the chart, if this is not a typo maybe remote the `-chart`.

Reply:
good catch (not that it matters in reality, but  you are right it's not consistent)